syft/frameworks/torch/tensors/interpreters/replicated_shared.py

```diff
@@ -59,8 +70,8 @@ def generate_shares(self, plain_text, number_of_shares=3):
     def __distribute_shares(workers, shares):
         shares_map = {}
         for i in range(len(shares)):
-            pointer1 = shares[i].send(workers[i])
-            pointer2 = shares[(i + 1) % len(shares)].send(workers[i])
+            pointer1 = shares[i].copy().send(workers[i])
+            pointer2 = shares[(i + 1) % len(shares)].copy().send(workers[i])
```

Comment:
Why is copy() needed, I thought copy() wasn't an inplace op?

Reply:
At one point when I send the same ```share[i]``` to 2 workers and it messes up with the garbage collection :\ (from my understanding this is because only one of the pointers should control the garbage collection - but sending it to two workers it means that one has the ```gc``` flag ```on``` and the one ```off```)

With the ```copy``` I make sure there are no changes in the process generated by the ```send``` method

Reply:
Ok this is weird indeed!